lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -209,6 +209,20 @@ def _install_if_needed(self, plugin: OpenSearchPlugin) -> bool:
 
         return self._install_plugin(plugin)
 
+    def configure(self, plugin: OpenSearchPlugin) -> bool:
```

Comment:
Could you explain why?

Reply:
`_configure_if_needed` does: (1) check IF we have a plugin installed but not configured and then does its work; and (2) internal to OpenSearchPlugin and its Manager.
Think about the case of moving from a GCP cloud to AWS. In this case, we *do* have an already configured plugin, but we want to reconfigure it nonetheless. That is why we need a method that removes/adds configuration disregard what is the current plugin status.

Reply:
It looks like `configure` and `disable` are simply proxy methods for `_apply_config` - it is safe to refer to `apply_config(...)` directly instead of having these 2 new methods